apps/web/src/pages/templates/editor/TemplateEditorPage.tsx

```diff
@@ -50,8 +54,32 @@ export default function TemplateEditorPage() {
     trigger,
     onTriggerModalDismiss,
   } = useTemplateController(templateId);
+
   const [showModal, confirmNavigation, cancelNavigation] = usePrompt(isDirty);
 
+  const [testWorkflowModalOpened, { close: closeTestWorkflowModal, open: openTestWorkflowModal }] = useDisclosure(
+    false,
+    {
+      onClose() {
+        onTestWorkflowDismiss();
+      },
+    }
+  );
+  const [saveChangesModalOpened, { close: closeSaveChangesModal, open: openSaveChangesModal }] = useDisclosure(false);
+
+  const onConfirmSaveChanges = async () => {
+    await handleSubmit(onSubmit)();
+    closeSaveChangesModal();
+    openTestWorkflowModal();
+  };
+
+  const onTestWorkflowClicked = async () => {
+    if (isDirty) {
+      openSaveChangesModal();
```

Comment:
should we return here?

Reply:
ohh right. oops. It opens one of the modals. I'll add an else. And remove the async